DataConnectors/ProofpointPOD/ProofpointSentinelConnector/__init__.py

```diff
@@ -0,0 +1,156 @@
+import os
+import datetime
+import socket
+import websocket
+import json
+import ssl
+import time
+import base64
+import hashlib
+import hmac
+import requests
+import azure.functions as func
+import logging
+
+
+customer_id = os.environ['WorkspaceID'] 
+shared_key = os.environ['WorkspaceKey']
+cluster_id = os.environ['ProofpointClusterID']
+_token = os.environ['ProofpointToken']
+time_delay_minutes = 60
+event_types = ["maillog","message"]
+
+def main(mytimer: func.TimerRequest) -> None:
+    if mytimer.past_due:
+        logging.info('The timer is past due!')
+
+    logging.info('Starting program')
+    if datetime.datetime.utcnow().minute < 2:
+        time.sleep(120)
+    api = Proofpoint_api()
+    for evt_type in event_types:
+        api.get_data(event_type=evt_type)
+
+class Proofpoint_api:
+    def __init__(self):
+        self.cluster_id = cluster_id
+        self._token = _token
+        self.time_delay_minutes = int(time_delay_minutes)
+        self.gen_timeframe(time_delay_minutes=self.time_delay_minutes)
+
+    def gen_timeframe(self, time_delay_minutes):
+        before_time = datetime.datetime.utcnow() - datetime.timedelta(minutes=time_delay_minutes)
+        self.before_time = before_time.strftime("%Y-%m-%dT%H:59:00.000000")
+        self.after_time = before_time.strftime("%Y-%m-%dT%H:01:00.000000")
+
+    def set_websocket_conn(self, event_type):
+        url = f"wss://logstream.proofpoint.com:443/v1/stream?cid={self.cluster_id}&type={event_type}&sinceTime={self.after_time}&toTime={self.before_time}"
+        logging.info('Opening Websocket logstream {}'.format(url))
+        # defining headers for websocket connection (do not change this)
+        header = {
+            "Connection": "Upgrade",
+            "Upgrade": "websocket",
+            "Authorization": f"Bearer {self._token}",
+            "Sec-WebSocket-Key": "SGVsbG8sIHdvcmxkIQ==",
+            "Sec-WebSocket-Version": "13"
+        }
+        try:
+            ws = websocket.create_connection(url, header=header, sslopt={"cert_reqs": ssl.CERT_NONE})
```

Comment:
Why set CERT_NONE, this will create potential security issues for customer?

Reply:
We can use ssl.CERT_REQUIRED. The question here is how to pass a valid set of CA certificates. Is there a way to point to some internal store that Function App framework has underneath?
Another option we can use - on Function App configuration step (aside of credentials) ask client to provide cert as string that he can get from browser (or any other tool) and then validate it within Function App while connecting to socket.

Reply:
is it SSL cert validation setting on client, why do you need to pass in a cert?

Reply:
Fixed, using certifi library of well-known CA issued certs